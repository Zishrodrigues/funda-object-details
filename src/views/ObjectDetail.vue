<template>
  <ErrorMessage
   v-if="error"
  >
    {{error}}
  </ErrorMessage>

  <Loader
    v-if="loading"
  />

  <template v-if="currentObject">
    <div class="flex border-orange border-solid border-0 border-b-05">
      <div class="flex-1">
        <Image
          :imageSrc="currentObject.HoofdFoto"
          :imageSrcSmall="currentObject.Media[0].MediaItems[0].Url"
          :imageSrcMedium="currentObject.Media[0].MediaItems[1].Url"
          :imageSrcLarge="currentObject.Media[0].MediaItems[2].Url"
          :alt="currentObject.HoofdFoto"
          classes="block h-full w-full object-cover"
        />
      </div>
      <div class="flex-1 p-1">
        <Title>
          <template v-slot:title>
            {{currentObject.Adres}}
          </template>
          <template v-slot:subTitle>
            {{currentObject.Postcode}} {{currentObject.Plaats}}
          </template>
        </Title>

        <DescriptionList classes="mt-05">
          <DescriptionListItem>
            <template v-slot:description-t>
              Asking price
            </template>
            <template v-slot:description-d>
              {{objectPrice}}
            </template>
          </DescriptionListItem>
          <DescriptionListItem>
            <template v-slot:description-t>
              Number of rooms
            </template>
            <template v-slot:description-d>
              {{currentObject.AantalKamers}}
            </template>
          </DescriptionListItem>
        </DescriptionList>
      </div>
    </div>

    <ImageGrid
      v-if="currentObjectImages"
      :images="currentObjectImages"
    />

    <NextButton
      v-on:click="nextObject"
    />
  </template>
</template>

<script>
import { onMounted, computed } from 'vue';

// components
import Image from '@/components/ui/Image.vue';
import ErrorMessage from '@/components/ui/ErrorMessage.vue';
import Loader from '@/components/ui/Loader.vue';
import Title from '@/components/ui/Title.vue';
import DescriptionList from '@/components/ui/DescriptionList.vue';
import DescriptionListItem from '@/components/ui/DescriptionListItem.vue';
import ImageGrid from '@/components/ImageGrid.vue';
import NextButton from '@/components/NextButton.vue';

// hooks
import { useObjectData } from '@/hooks/useObjectData';

// services
import { formatPrice, getRandomObjectId } from '@/services/helpers';

export default {
  name: 'Object Detail',
  components: {
    Image,
    ImageGrid,
    ErrorMessage,
    Loader,
    Title,
    DescriptionList,
    DescriptionListItem,
    NextButton
  },
  setup() {
    const {
      currentObject,
      currentObjectImages,
      getObject,
      loading,
      error
    } = useObjectData();

    onMounted(() => {
      getObject();
    });

    const nextObject = () => {
      const id = getRandomObjectId(currentObject.value.InternalId);
      getObject(id);
    };

    const objectPrice = computed(() => {
      return formatPrice(currentObject.value.Koopprijs);
    });

    return {
      currentObject,
      currentObjectImages,
      objectPrice,
      nextObject,
      loading,
      error
    };
  }
};
</script>
