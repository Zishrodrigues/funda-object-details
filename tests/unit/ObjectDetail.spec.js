import { shallowMount, mount } from '@vue/test-utils';
import ObjectDetail from '@/views/ObjectDetail.vue';

describe('ObjectDetail.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(ObjectDetail);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Should not display an error message as default', async() => {
    const wrapper = shallowMount(ObjectDetail);
    const DOM = wrapper.find('.error');
    expect(DOM.exists()).toBe(false);
  });

  it('Should not be loading on default', async() => {
    const wrapper = shallowMount(ObjectDetail);
    const DOM = wrapper.find('.loading');
    expect(DOM.exists()).toBe(false);
  });
});
