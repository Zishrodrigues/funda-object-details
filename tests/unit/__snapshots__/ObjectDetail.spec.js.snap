// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectDetail.vue renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  
</div>
`;
